Build failure with message "error: no such module 'Zip'"

When I am trying to build the package locally I get the following error:

`
Compiling VaporWalletOrders OrdersServiceCustom.swift
/Users/runner/work/wallet/wallet/Sources/VaporWalletOrders/OrdersServiceCustom.swift:10:8: error: no such module 'Zip'
  8 | import VaporWallet
  9 | import WalletOrders
 10 | import Zip
    |        `- error: no such module 'Zip'`

I have fork the repo and run the Github action and get the same error https://github.com/csknns/wallet/actions/runs/14143509636/job/39627981687?pr=2.

The last Github Action on this repo seems to pass. Adding the zip dependency in the package manifest does not seem to fix the issue because I get missing argument errors for the ZIP methods.

Do you have any ideas on what might be the problem? Perhaps it builds in the CI because it picks the Zip module as a transient dependency but I would not find it in the dependency tree anywhere. 
Thanks!

--- a/Package.swift
+++ b/Package.swift
@@ -15,6 +15,7 @@ let package = Package(
         .package(url: "https://github.com/vapor/fluent.git", from: "4.12.0"),
         .package(url: "https://github.com/fpseverino/fluent-wallet.git", from: "0.1.0"),
         .package(url: "https://github.com/vapor/apns.git", from: "4.2.0"),
+        .package(url: "https://github.com/adam-fowler/swift-zip-archive.git", from: "0.6.3"),
         // used in tests
         .package(url: "https://github.com/vapor/fluent-sqlite-driver.git", from: "4.8.0"),
     ],
@@ -34,6 +35,7 @@ let package = Package(
             dependencies: [
                 .target(name: "VaporWallet"),
                 .product(name: "FluentWalletPasses", package: "fluent-wallet"),
+                .product(name: "ZipArchive", package: "swift-zip-archive"),
             ],
             swiftSettings: swiftSettings
         ),
@@ -43,6 +45,7 @@ let package = Package(
                 .target(name: "VaporWalletPasses"),
                 .product(name: "VaporTesting", package: "vapor"),
                 .product(name: "FluentSQLiteDriver", package: "fluent-sqlite-driver"),
+                .product(name: "ZipArchive", package: "swift-zip-archive"),
             ],
             resources: [
                 .copy("SourceFiles")
@@ -64,6 +67,7 @@ let package = Package(
                 .target(name: "VaporWalletOrders"),
                 .product(name: "VaporTesting", package: "vapor"),
                 .product(name: "FluentSQLiteDriver", package: "fluent-sqlite-driver"),
+                .product(name: "ZipArchive", package: "swift-zip-archive"),
             ],
             resources: [
                 .copy("SourceFiles")

--- a/Sources/VaporWalletOrders/OrdersServiceCustom.swift
+++ b/Sources/VaporWalletOrders/OrdersServiceCustom.swift
@@ -7,7 +7,6 @@ import Vapor
 import VaporAPNS
 import VaporWallet
 import WalletOrders
-import Zip
 
 /// Class to handle ``OrdersService``.
 ///

--- a/Sources/VaporWalletPasses/PassesServiceCustom.swift
+++ b/Sources/VaporWalletPasses/PassesServiceCustom.swift
@@ -7,7 +7,7 @@ import Vapor
 import VaporAPNS
 import VaporWallet
 import WalletPasses
-import Zip
+import ZipArchive
 
 /// Class to handle ``PassesService``.
 ///
@@ -172,13 +172,10 @@ extension PassesServiceCustom {
             throw WalletPassesError.invalidNumberOfPasses
         }
 
-        var files: [ArchiveFile] = []
+        let writer = ZipArchiveWriter()
         for (i, pass) in passes.enumerated() {
-            try await files.append(ArchiveFile(filename: "pass\(i).pkpass", data: self.build(pass: pass, on: db)))
+            try await writer.writeFile(filename: "pass\(i).pkpass", contents: Array(self.build(pass: pass, on: db)))
         }
-
-        let zipFile = FileManager.default.temporaryDirectory.appendingPathComponent("\(UUID().uuidString).pkpass")
-        try Zip.zipData(archiveFiles: files, zipFilePath: zipFile)
-        return try Data(contentsOf: zipFile)
+        return try Data(writer.finalizeBuffer())
     }
 }

--- a/Tests/VaporWalletOrdersTests/Utils/withApp.swift
+++ b/Tests/VaporWalletOrdersTests/Utils/withApp.swift
@@ -5,7 +5,6 @@ import Testing
 import Vapor
 import VaporWalletOrders
 import WalletOrders
-import Zip
 
 func withApp(
     useEncryptedKey: Bool = false,
@@ -31,8 +30,6 @@ func withApp(
         app.databases.middleware.use(ordersService, on: .sqlite)
 
         try app.grouped("api", "orders").register(collection: ordersService)
-
-        Zip.addCustomFileExtension("order")
 
         try await body(app, ordersService)
 

--- a/Tests/VaporWalletOrdersTests/VaporWalletOrdersTests.swift
+++ b/Tests/VaporWalletOrdersTests/VaporWalletOrdersTests.swift
@@ -2,7 +2,7 @@ import FluentWalletOrders
 import Testing
 import VaporTesting
 import WalletOrders
-import Zip
+import ZipArchive
 
 @testable import VaporWalletOrders
 
@@ -18,29 +18,24 @@ struct VaporWalletOrdersTests {
             try await orderData.create(on: app.db)
             let order = try await orderData.$order.get(on: app.db)
 
-            let data = try await ordersService.build(order: orderData, on: app.db)
+            let reader = try await ZipArchiveReader(buffer: ordersService.build(order: orderData, on: app.db))
+            let directory = try reader.readDirectory()
 
-            let orderURL = FileManager.default.temporaryDirectory.appendingPathComponent("\(UUID().uuidString).order")
-            try data.write(to: orderURL)
-            let orderFolder = FileManager.default.temporaryDirectory.appendingPathComponent(UUID().uuidString, isDirectory: true)
-            try Zip.unzipFile(orderURL, destination: orderFolder)
+            #expect(directory.contains { $0.filename == "signature" })
 
-            #expect(FileManager.default.fileExists(atPath: orderFolder.path.appending("/signature")))
+            #expect(directory.contains { $0.filename == "pet_store_logo.png" })
+            #expect(directory.contains { $0.filename == "it-IT.lproj/pet_store_logo.png" })
 
-            #expect(FileManager.default.fileExists(atPath: orderFolder.path.appending("/pet_store_logo.png")))
-            #expect(FileManager.default.fileExists(atPath: orderFolder.path.appending("/it-IT.lproj/pet_store_logo.png")))
-
-            #expect(FileManager.default.fileExists(atPath: orderFolder.path.appending("/order.json")))
-            let orderJSONData = try String(contentsOfFile: orderFolder.path.appending("/order.json")).data(using: .utf8)
-            let orderJSON = try decoder.decode(OrderJSONData.self, from: orderJSONData!)
-            #expect(orderJSON.authenticationToken == order.authenticationToken)
+            let orderBytes = try reader.readFile(#require(directory.first { $0.filename == "order.json" }))
+            let roundTrippedOrder = try decoder.decode(OrderJSONData.self, from: Data(orderBytes))
+            #expect(roundTrippedOrder.authenticationToken == order.authenticationToken)
             let orderID = try order.requireID().uuidString
-            #expect(orderJSON.orderIdentifier == orderID)
+            #expect(roundTrippedOrder.orderIdentifier == orderID)
 
-            let manifestJSONData = try String(contentsOfFile: orderFolder.path.appending("/manifest.json")).data(using: .utf8)
-            let manifestJSON = try decoder.decode([String: String].self, from: manifestJSONData!)
-            let iconData = try Data(contentsOf: orderFolder.appendingPathComponent("/icon.png"))
-            #expect(manifestJSON["icon.png"] == SHA256.hash(data: iconData).hex)
+            let manifestJSONBytes = try reader.readFile(#require(directory.first { $0.filename == "manifest.json" }))
+            let manifestJSON = try decoder.decode([String: String].self, from: Data(manifestJSONBytes))
+            let iconBytes = try reader.readFile(#require(directory.first { $0.filename == "icon.png" }))
+            #expect(manifestJSON["icon.png"] == SHA256.hash(data: iconBytes).hex)
             #expect(manifestJSON["pet_store_logo.png"] != nil)
             #expect(manifestJSON["it-IT.lproj/pet_store_logo.png"] != nil)
         }

--- a/Tests/VaporWalletPassesTests/Utils/withApp.swift
+++ b/Tests/VaporWalletPassesTests/Utils/withApp.swift
@@ -5,7 +5,6 @@ import Testing
 import Vapor
 import VaporWalletPasses
 import WalletPasses
-import Zip
 
 func withApp(
     useEncryptedKey: Bool = false,
@@ -31,8 +30,6 @@ func withApp(
         app.databases.middleware.use(passesService, on: .sqlite)
 
         try app.grouped("api", "passes").register(collection: passesService)
-
-        Zip.addCustomFileExtension("pkpass")
 
         try await body(app, passesService)
 

--- a/Tests/VaporWalletPassesTests/VaporWalletPassesTests.swift
+++ b/Tests/VaporWalletPassesTests/VaporWalletPassesTests.swift
@@ -2,7 +2,7 @@ import FluentWalletPasses
 import Testing
 import VaporTesting
 import WalletPasses
-import Zip
+import ZipArchive
 
 @testable import VaporWalletPasses
 
@@ -18,32 +18,27 @@ struct VaporWalletPassesTests {
             try await passData.create(on: app.db)
             let pass = try await passData.$pass.get(on: app.db)
 
-            let data = try await passesService.build(pass: passData, on: app.db)
+            let reader = try await ZipArchiveReader(buffer: passesService.build(pass: passData, on: app.db))
+            let directory = try reader.readDirectory()
 
-            let passURL = FileManager.default.temporaryDirectory.appendingPathComponent("\(UUID().uuidString).pkpass")
-            try data.write(to: passURL)
-            let passFolder = FileManager.default.temporaryDirectory.appendingPathComponent(UUID().uuidString, isDirectory: true)
-            try Zip.unzipFile(passURL, destination: passFolder)
+            #expect(directory.contains { $0.filename == "signature" })
 
-            #expect(FileManager.default.fileExists(atPath: passFolder.path.appending("/signature")))
+            #expect(directory.contains { $0.filename == "logo.png" })
+            #expect(directory.contains { $0.filename == "personalizationLogo.png" })
+            #expect(directory.contains { $0.filename == "it-IT.lproj/logo.png" })
+            #expect(directory.contains { $0.filename == "it-IT.lproj/personalizationLogo.png" })
 
-            #expect(FileManager.default.fileExists(atPath: passFolder.path.appending("/logo.png")))
-            #expect(FileManager.default.fileExists(atPath: passFolder.path.appending("/personalizationLogo.png")))
-            #expect(FileManager.default.fileExists(atPath: passFolder.path.appending("/it-IT.lproj/logo.png")))
-            #expect(FileManager.default.fileExists(atPath: passFolder.path.appending("/it-IT.lproj/personalizationLogo.png")))
-
-            #expect(FileManager.default.fileExists(atPath: passFolder.path.appending("/pass.json")))
-            let passJSONData = try String(contentsOfFile: passFolder.path.appending("/pass.json")).data(using: .utf8)
-            let passJSON = try decoder.decode(PassJSONData.self, from: passJSONData!)
-            #expect(passJSON.authenticationToken == pass.authenticationToken)
+            let passBytes = try reader.readFile(#require(directory.first { $0.filename == "pass.json" }))
+            let roundTrippedPass = try decoder.decode(PassJSONData.self, from: Data(passBytes))
+            #expect(roundTrippedPass.authenticationToken == pass.authenticationToken)
             let passID = try pass.requireID().uuidString
-            #expect(passJSON.serialNumber == passID)
-            #expect(passJSON.description == passData.title)
+            #expect(roundTrippedPass.serialNumber == passID)
+            #expect(roundTrippedPass.description == passData.title)
 
-            let manifestJSONData = try String(contentsOfFile: passFolder.path.appending("/manifest.json")).data(using: .utf8)
-            let manifestJSON = try decoder.decode([String: String].self, from: manifestJSONData!)
-            let iconData = try Data(contentsOf: passFolder.appendingPathComponent("/icon.png"))
-            #expect(manifestJSON["icon.png"] == Insecure.SHA1.hash(data: iconData).hex)
+            let manifestJSONBytes = try reader.readFile(#require(directory.first { $0.filename == "manifest.json" }))
+            let manifestJSON = try decoder.decode([String: String].self, from: Data(manifestJSONBytes))
+            let iconBytes = try reader.readFile(#require(directory.first { $0.filename == "icon.png" }))
+            #expect(manifestJSON["icon.png"] == Insecure.SHA1.hash(data: iconBytes).hex)
             #expect(manifestJSON["logo.png"] != nil)
             #expect(manifestJSON["personalizationLogo.png"] != nil)
             #expect(manifestJSON["it-IT.lproj/logo.png"] != nil)
@@ -79,36 +74,31 @@ struct VaporWalletPassesTests {
             try await passData.create(on: app.db)
             let pass = try await passData.$pass.get(on: app.db)
 
-            let data = try await passesService.build(pass: passData, on: app.db)
+            let reader = try await ZipArchiveReader(buffer: passesService.build(pass: passData, on: app.db))
+            let directory = try reader.readDirectory()
 
-            let passURL = FileManager.default.temporaryDirectory.appendingPathComponent("\(UUID().uuidString).pkpass")
-            try data.write(to: passURL)
-            let passFolder = FileManager.default.temporaryDirectory.appendingPathComponent(UUID().uuidString, isDirectory: true)
-            try Zip.unzipFile(passURL, destination: passFolder)
+            #expect(directory.contains { $0.filename == "signature" })
 
-            #expect(FileManager.default.fileExists(atPath: passFolder.path.appending("/signature")))
+            #expect(directory.contains { $0.filename == "logo.png" })
+            #expect(directory.contains { $0.filename == "personalizationLogo.png" })
+            #expect(directory.contains { $0.filename == "it-IT.lproj/logo.png" })
+            #expect(directory.contains { $0.filename == "it-IT.lproj/personalizationLogo.png" })
 
-            #expect(FileManager.default.fileExists(atPath: passFolder.path.appending("/logo.png")))
-            #expect(FileManager.default.fileExists(atPath: passFolder.path.appending("/personalizationLogo.png")))
-            #expect(FileManager.default.fileExists(atPath: passFolder.path.appending("/it-IT.lproj/logo.png")))
-            #expect(FileManager.default.fileExists(atPath: passFolder.path.appending("/it-IT.lproj/personalizationLogo.png")))
-
-            #expect(FileManager.default.fileExists(atPath: passFolder.path.appending("/pass.json")))
-            let passJSONData = try String(contentsOfFile: passFolder.path.appending("/pass.json")).data(using: .utf8)
-            let passJSON = try decoder.decode(PassJSONData.self, from: passJSONData!)
-            #expect(passJSON.authenticationToken == pass.authenticationToken)
+            let passBytes = try reader.readFile(#require(directory.first { $0.filename == "pass.json" }))
+            let roundTrippedPass = try decoder.decode(PassJSONData.self, from: Data(passBytes))
+            #expect(roundTrippedPass.authenticationToken == pass.authenticationToken)
             let passID = try pass.requireID().uuidString
-            #expect(passJSON.serialNumber == passID)
-            #expect(passJSON.description == passData.title)
+            #expect(roundTrippedPass.serialNumber == passID)
+            #expect(roundTrippedPass.description == passData.title)
 
-            let personalizationJSONData = try String(contentsOfFile: passFolder.path.appending("/personalization.json")).data(using: .utf8)
-            let personalizationJSON = try decoder.decode(PersonalizationJSON.self, from: personalizationJSONData!)
+            let personalizationJSONBytes = try reader.readFile(#require(directory.first { $0.filename == "personalization.json" }))
+            let personalizationJSON = try decoder.decode(PersonalizationJSON.self, from: Data(personalizationJSONBytes))
             #expect(personalizationJSON.description == "Hello, World!")
 
-            let manifestJSONData = try String(contentsOfFile: passFolder.path.appending("/manifest.json")).data(using: .utf8)
-            let manifestJSON = try decoder.decode([String: String].self, from: manifestJSONData!)
-            let personalizationLogoData = try Data(contentsOf: passFolder.appendingPathComponent("/personalizationLogo.png"))
-            let personalizationLogoHash = Insecure.SHA1.hash(data: personalizationLogoData).hex
+            let manifestJSONBytes = try reader.readFile(#require(directory.first { $0.filename == "manifest.json" }))
+            let manifestJSON = try decoder.decode([String: String].self, from: Data(manifestJSONBytes))
+            let personalizationLogoBytes = try reader.readFile(#require(directory.first { $0.filename == "personalizationLogo.png" }))
+            let personalizationLogoHash = Insecure.SHA1.hash(data: personalizationLogoBytes).hex
             #expect(manifestJSON["personalizationLogo.png"] == personalizationLogoHash)
             #expect(manifestJSON["it-IT.lproj/personalizationLogo.png"] == personalizationLogoHash)
         }

